tmt/steps/provision/__init__.py

```diff
@@ -1136,21 +1146,29 @@ def _sanitize_ansible_playbook_path(
             the eventual playbook path is not absolute.
         """
 
-        # Some playbooks must be under playbook root, which is often
-        # a metadata tree root.
-        if playbook_root is not None:
-            playbook = playbook_root / playbook.unrooted()
+        if isinstance(playbook, Path):
+            # Some playbooks must be under playbook root, which is often
+            # a metadata tree root.
+            if playbook_root is not None:
+                playbook = playbook_root / playbook.unrooted()
 
-            if not playbook.is_relative_to(playbook_root):
-                raise tmt.utils.GeneralError(
-                    f"'{playbook}' is not relative to the expected root '{playbook_root}'.")
+                if not playbook.is_relative_to(playbook_root):
+                    raise tmt.utils.GeneralError(
+                        f"'{playbook}' is not relative to the expected root '{playbook_root}'.")
+
+            if not playbook.exists():
+                raise tmt.utils.FileError(f"Playbook '{playbook}' does not exist.")
+
+            self.debug(f"Playbook full path: '{playbook}'", level=2)
+
+            return playbook
 
-        if not playbook.exists():
-            raise tmt.utils.FileError(f"Playbook '{playbook}' does not exist.")
+        if isinstance(playbook, str):
```

Comment:
I would add here a comment this is for the `AnsibleCollectionPlaybook` type

```suggestion
        # AnsibleCollectionPlaybook
        if isinstance(playbook, str):
```

Reply:
Addressed in ceb00541